### ISSUE TITLE
Update UTBotJava project to Gradle 7.0+

### ISSUE THREAD

**Issue by alisevych** [UnitTestBot/UTBotJava#265]
**Description**

Update UTBotJava project to work with Gradle 7.0+

**Expected behavior**

UTBotJava project can be successfully built using Gradle 7.0+

**Actual behavior**

Currently Gradle version is set to 6.8 in gradle/wrapper/gradle-wrapper.properties
There are warnings regarding deprecated features: `This will fail with an error in Gradle 7.0.`
When using specified Gradle 7.4.2 the build fails with errors (please see the details in Context).

**Environment**

Open UTBotJava as Gradle project in IntelliJ IDEA.
One can change the defined Gradle wrapper to Gradle from specified location:
Gradle settings -> Use Gradle from: Specified location : select Gradle 7.4.2 home

**Context**

- With Gradle 6.8:
When building with --warning-mode all there are following warnings in log:

```
Configure project :utbot-framework
The testCompile configuration has been deprecated for dependency declaration. This will fail with an error in Gradle 7.0. Please use the testImplementation configuration instead. Consult the upgrading guide for further information: https://docs.gradle.org/6.8/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations
	at build_798bvsh9ns8bbholc7dhjgky9$_run_closure3.doCall(C:\Users\lWX1182794\UTBotJava\utbot-framework\build.gradle:52)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)

Configure project :utbot-analytics
Using force on a dependency has been deprecated. This is scheduled to be removed in Gradle 7.0. Consider using strict version constraints instead (version { strictly ... } }). Consult the upgrading guide for further information: https://docs.gradle.org/6.8/userguide/upgrading_version_5.html#forced_dependencies
	at jvm_project_acg0tf5btrekvhk3jw6xoornm$_run_closure1$_closure8.doCall(C:\Users\lWX1182794\UTBotJava\gradle\include\jvm-project.gradle:11)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)

Configure project :utbot-cli
The compile configuration has been deprecated for dependency declaration. This will fail with an error in Gradle 7.0. Please use the implementation configuration instead. Consult the upgrading guide for further information: https://docs.gradle.org/6.8/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations
	at build_aosn2mtnaytu1oe84nqqhjzs4$_run_closure2.doCall(C:\Users\lWX1182794\UTBotJava\utbot-cli\build.gradle:25)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
The AbstractArchiveTask.version property has been deprecated. This is scheduled to be removed in Gradle 7.0. Please use the archiveVersion property instead. See https://docs.gradle.org/6.8/dsl/org.gradle.api.tasks.bundling.AbstractArchiveTask.html#org.gradle.api.tasks.bundling.AbstractArchiveTask:version for more details.
	at build_aosn2mtnaytu1oe84nqqhjzs4$_run_closure5.doCall(C:\Users\lWX1182794\UTBotJava\utbot-cli\build.gradle:53)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
```

- When using specified Gradle 7.4.2 the build is failing with errors. For example:

```
Exception is:
org.gradle.api.plugins.UnknownPluginException: Plugin [id: 'maven'] was not found in any of the following sources:

A problem occurred evaluating project ':utbot-maven'.
> Could not get unknown property 'install' for project ':utbot-maven' of type org.gradle.api.Project.

A problem occurred evaluating project ':utbot-sample'.
> Could not find method compile() for arguments [{group=com.github.stephenc.findbugs, name=findbugs-annotations, version=1.3.9-1}] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.
```

**Comment by korifey** [comment]
I think we must simultaneously update to new gradle and Kotlin

**Comment by alisevych** [comment]
Closing as this issue is included into #749
